### PR TITLE
store/tikv: move batch_getter from tikv package to driver

### PR DESCRIPTION
**Pull request by AndreMouche**
Signed-off-by: AndreMouche

### What problem does this PR solve?
This PR moves BatchGetter from tikv package to tikv_driver so we needn't convert to tikv interface when using BatchGetter.
Part of https://github.com/pingcap/tidb/issues/22513

### Check List 
Tests 
- Unit test

### Release note 
- No release note

### CODE REVIEW

**Review by disksing — APPROVED** [review]

File: store/driver/txn/batch_getter.go [hunk 1]
```diff
@@ -52,23 +52,23 @@ func NewBufferBatchGetter(buffer BatchBufferGetter, middleCache Getter, snapshot
...
+func (b *BufferBatchGetter) BatchGet(ctx context.Context, keys []kv.Key) (map[string][]byte, error) {
 	if b.buffer.Len() == 0 {
 		return b.snapshot.BatchGet(ctx, keys)
 	}
 	bufferValues := make([][]byte, len(keys))
-	shrinkKeys := make([][]byte, 0, len(keys))
+	shrinkKeys := make([]kv.Key, 0, len(keys))
 	for i, key := range keys {
-		val, err := b.buffer.Get(key)
+		val, err := b.buffer.Get(context.Background(), key)
```
Comment (sticnarf) [inline store/driver/txn/batch_getter.go:62]: Why using background instead of passing `ctx` in?

File: store/driver/txn/batch_getter.go [hunk 2]
```diff
@@ -52,23 +52,23 @@ func NewBufferBatchGetter(buffer BatchBufferGetter, middleCache Getter, snapshot
...
 			bufferValues[i] = val
 			continue
 		}
-		if !tidbkv.IsErrNotFound(err) {
+		if !kv.IsErrNotFound(err) {
 			return nil, errors.Trace(err)
 		}
 		if b.middle != nil {
-			val, err = b.middle.Get(key)
+			val, err = b.middle.Get(context.Background(), key)
```
Comment (sticnarf) [inline store/driver/txn/batch_getter.go:71]: and ditto

File: store/driver/txn/batch_getter.go [hunk 3]
```diff
@@ -29,14 +29,14 @@ type BatchBufferGetter interface {
...
-	BatchGet(ctx context.Context, keys [][]byte) (map[string][]byte, error)
+	BatchGet(ctx context.Context, keys []kv.Key) (map[string][]byte, error)
 }
 
 // Getter is the interface for the Get method.
 type Getter interface {
 	// Get gets the value for key k from kv store.
 	// If corresponding kv pair does not exist, it returns nil and ErrNotExist.
-	Get(k []byte) ([]byte, error)
+	Get(_ context.Context, k kv.Key) ([]byte, error)
```
Comment (youjiali1995) [inline store/driver/txn/batch_getter.go:39]:
```suggestion
	Get(ctx context.Context, k kv.Key) ([]byte, error)
```

File: store/tikv/tests/util_test.go [hunk 4]
```diff
@@ -114,3 +117,12 @@ func valueBytes(n int) []byte {
 func s08d(prefix string, n int) string {
 	return fmt.Sprintf("%s%08d", prefix, n)
 }
+
+func toTiDBTxn(txn *tikv.TxnProbe) kv.Transaction {
+	return txndriver.NewTiKVTxn(txn.KVTxn)
+}
```
Comment (youjiali1995) [inline store/tikv/tests/util_test.go:123]: I'm confused the function name is `toTiDBTxn`, but it calls `NewTiKVTxn`.

File: store/driver/txn/batch_getter.go (hunk 2 above)
Comment (AndreMouche) [reply store/driver/txn/batch_getter.go]: nice catch!

File: store/tikv/tests/util_test.go (hunk 4 above)
Comment (AndreMouche) [reply store/tikv/tests/util_test.go:123]: Here we need a `tidb/kv.Transaction`, however, I have no idea about the function name

**Review by youjiali1995 — APPROVED** [review]
LGTM

**Pull request merged**